redislock.go

```diff
@@ -24,30 +25,31 @@ var (
 
 // NewRedisLocker provides an implementation of the Locker interface using
 // redis for storage.
-func NewRedisLocker(r redis.UniversalClient, options ...redsync.Option) (gocron.Locker, error) {
+func NewRedisLocker(r redis.UniversalClient, autoExtendDuration time.Duration, options ...redsync.Option) (gocron.Locker, error) {
```

Comment:
this is a breaking change (so let's not do it this way). An oversight on my part to have the options be ...resync.Option instead of a gocron.RedisLockerOption, to which we could just add WithAutoExtendDuration.

Instead, let's do it the way it should have been:
- Add a new method `func NewRedisLockerWithOptions(r redis.UniversalClient, options ...LockerOption) (gocron.Locker, error)`
- Define the LockerOption as `type LockerOption func(*redisLocker) error`
- Then, we could define all the options, with `Option` appended to the name so they are unique, as LockerOption type
- Then, also add `WithAutoExtendDuration` as an Option and implement it through there.

Let me know if you would like to complete this work. If not, I will take it on when I have time.

Reply:
I made some changes:
1. Add `LockOption`, support `WithAutoExtendDuration`,`WithRedsyncOptions` to be compatible with redsync options
2. Added `NewRedisLockerWithOptions`
3. Modified `TestAutoExtend` 

Please review it again.